### ISSUE TITLE
Add mock provider level resource overrides

### ISSUE THREAD

**Issue by wyardley** [opentofu/opentofu#2048]
### OpenTofu Version

```shell
OpenTofu v1.8.2
on darwin_arm64
+ provider registry.opentofu.org/hashicorp/random v3.6.3
```

### The problem in your OpenTofu project

While currently tofu (after #1204) supports resource overrides at top level and within run blocks, as mentioned there, it does not yet support mock provider level overrides.

Since this is supported in Terraform, it would be nice to have feature parity with Terraform

### Attempted Solutions

```hcl
mock_provider "random" {
  override_resource {
    target = random_integer.count
    values = {
      result = 58
    }
  }
  alias = "fake"
}
```

This returns the error
```
│ Error: Unsupported block type
│ 
│   on main.tftest.hcl line 2, in mock_provider "random":
│    2:   override_resource {
│ 
│ Blocks of type "override_resource" are not expected here.
╵
```

**Comment by cam72cam** [comment]
Thanks for the issue, we decided to see what the community's interest in this feature is before scheduling it as `test` is a relatively new feature.  I would be interested to know what modules/projects are using this feature, that would help us gauge the level of adoption.

That said, we will triage this in the core team next week and likely mark it as accepted, waiting on a community PR.
